crates/bevy_state/src/app.rs

```diff
@@ -120,6 +132,14 @@ impl AppExtStates for SubApp {
 
         self
     }
+
+    #[cfg(feature = "bevy_hierarchy")]
+    fn enable_state_scoped_entities<S: States>(&mut self) -> &mut Self {
+        self.add_systems(
+            StateTransition,
+            clear_state_scoped_entities::<S>.in_set(StateTransitionSteps::ExitSchedules),
+        )
```

Comment:
This should warn if the app doesn't have `S` as a state already, or add it anyway